Go: Refine Initialize Metrics (#3305)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/go/manual.md

)

func main() {

>   // Create resource.
> 	res, err := newResource()
> 	if err != nil {
> 		panic(err)

Reviewer: ```suggestion
  	// Create resource.
	res, err := newResource()
	if err != nil {
		panic(err)
```

	}

	// Create a meter provider.
	// You can pass this instance directly to your instrumented code if it
	// accepts a MeterProvider instance.
	meterProvider, err := newMeterProvider(res)
	if err != nil {
		panic(err)
	}

	// Handle shutdown properly so nothing leaks.
	defer func() {
		if err := meterProvider.Shutdown(context.Background()); err != nil {
			log.Println(err)
		}
	}()

	// Register as global meter provider so that it can

> 	// that can used via otel.Meter and accessed using otel.GetMeterProvider.

Reviewer 2: ```suggestion
	// then be used via otel.Meter and accessed using otel.GetMeterProvider.
```

Author: https://github.com/open-telemetry/opentelemetry.io/pull/3305/commits/45d6031ac453425eca05c9bc52d19f4d28086d5e

	// Most instrumentation libraries use the global meter provider as default.
	// If the global meter provider is not set then a no-op implementation

> 	// is used and which fails to generate data.

Reviewer 2: ```suggestion
	// is used, which fails to generate data.
```

Author: https://github.com/open-telemetry/opentelemetry.io/pull/3305/commits/45d6031ac453425eca05c9bc52d19f4d28086d5e
